POWERSYNC: add migration to create postgresql publication

Dieterbe opened the pull request.

rolandgeider committed:
Update react files and set type="module"

This is needed since the move to vite on the react side

Dieterbe committed:
add migration to create postgresql publication

Dieterbe commented on wger/nutrition/migrations/0025_create_publication.py:
```diff
@@ -0,0 +1,26 @@
+                IF NOT EXISTS (
+                    SELECT 1 FROM pg_publication WHERE pubname = 'powersync'
+                ) THEN
+                    CREATE PUBLICATION powersync FOR TABLE nutrition_ingredient, auth_user;
```
these tables are for testing purpose. will refine later

rolandgeider committed:
Add URL to get token for powersync

rolandgeider committed:
Merge branch 'refs/heads/master' into powersync

rolandgeider committed:
Return powersync public key(s)

rolandgeider committed:
Don't use the login_required decorator

This wasn't working well with when used over the API

rolandgeider committed:
Fix the powersync key generation

This still needs to be clean up (specially the key generation, at the moment we
use a new dependency, but all this can be also done with the library we currently
use)

rolandgeider committed:
Install python-jose via simplejwt

Dieterbe committed:
just create the publication for all tables.

most convenient and should be just fine

rolandgeider committed:
Allow migration to run on non-postgres databases

Dieterbe committed:
todo

rolandgeider committed:
Add foreign key to the user table

This is only needed because powersync can't handle joins

rolandgeider committed:
Revert "Add foreign key to the user table"

This reverts commit c6163de979c61db10109bfb86cc3d870281b6ad6.

rolandgeider committed:
Split migrations

- move create publication to core
- use ivm entries for the nutrition publications

rolandgeider committed:
Merge branch 'master' into powersync

# Conflicts:
#	wger/core/static/react/main.js

rolandgeider committed:
Add development Dockerfile with additional packages

Dieterbe committed:
give dummy meals names

rolandgeider committed:
Add UUIDs to some of the nutrition models

This is needed in the flutter app to properly be able to set IDs on device, since
the regular IDs can only be set by postgres.

Dieterbe committed:
also translate foreign keys to uuid's

rolandgeider committed:
Merge branch 'master' into powersync

# Conflicts:
#	extras/docker/development/Dockerfile
#	extras/docker/production/settings.py
#	requirements.txt
#	wger/core/api/views.py
#	wger/nutrition/models/plan.py
#	wger/urls.py

rolandgeider committed:
Fix optional dependency

rolandgeider committed:
Fix migration

rolandgeider committed:
Correctly process weigh entry data as sent by powersync

rolandgeider committed:
Add powersync logic to handle workout logs

rolandgeider committed:
Add powersync logic to handle workout sessions

rolandgeider committed:
Pass the request to the powersync helpers

rolandgeider committed:
Rename migration

rolandgeider committed:
Merge branch 'master' into powersync

# Conflicts:
#	extras/docker/production/settings.py
#	pyproject.toml
#	uv.lock
#	wger/weight/models.py

rolandgeider committed:
Add powersync url to settings file

rolandgeider committed:
Cleanup migrations

We don't need the ivm extension

rolandgeider committed:
Add UUIDs to measurement models

rolandgeider committed:
Commit updated uv.lock

rolandgeider committed:
Fix type

rolandgeider committed:
Fix typo

rolandgeider committed:
Change primary key type for Category and Measurement models to UUID

rolandgeider committed:
Change primary key type for WorkoutLog and Session models to UUID

rolandgeider committed:
Merge branch 'master' into powersync

# Conflicts:
#	CHANGELOG.md
#	pyproject.toml
#	settings/main.py
#	uv.lock
#	wger/manager/models/log.py

rolandgeider committed:
Return errors to the powersync client

rolandgeider committed:
Fix migrations

rolandgeider committed:
Centralize owner checking logic

rolandgeider commented:
hey @real-yfprojects, sorry for the late reply, hadn't seen your message.

Most of the work will need to happen on the flutter side of things, but here there will be some cleaning up needed to do before we can really release this (we also moved some models to use uuids as the PK so we can generate them client side as well without having to do mappings back and forth, so possibly something among those lines as well)

rolandgeider committed:
Add caps for progression calculations to prevent overflow

rolandgeider committed:
Merge branch 'master' into powersync

rolandgeider committed:
Split migrations so they run in individual transactions

rolandgeider committed:
Also migrate the data stored in the json field for context data

rolandgeider committed:
Simplify generation of JWT keys for powersync

rolandgeider committed:
Add powersync JWT key generator

rolandgeider committed:
Add check for default powersync keys

rolandgeider committed:
Add management command to prepare the powersync storage

rolandgeider committed:
Cast the userId as a string

Otherwise, the type mismatch can cause the sync logic in powersync to not
find the user's data

rolandgeider committed:
Add new POWERSYNC_URL_PATH config option

This will be the default for mostly everyone since the powersync service
will be behind a reverse proxy. For other settings (or during development)
POWERSYNC_URL is still available.

rolandgeider committed:
Fix tests

rolandgeider committed:
Cleanup

rolandgeider committed:
Change measurement entry default to be timezone-aware

rolandgeider committed:
Return error codes to the client

rolandgeider committed:
Handle routine and nutritional plans edit operations

rolandgeider committed:
Refactor use gating in WorkoutLogs

Now it's possible to pass an explicit session ID (this field was read-only
before).

rolandgeider committed:
Add edit operations for nutrition logs

rolandgeider committed:
Handle gallery edit operations

rolandgeider committed:
Add tests for the powersync CRUD endpoint

rolandgeider committed:
Sync the rest of the nutrition tables to powersync

rolandgeider committed:
Make the nutrition tables IDs to UUIDs

This makes using these tables much easier on the flutter side with powersync,
since there's no dancing and mapping around with UUIDs <--> IDs etc. The only
downside is that external clients that access these will need to update.

rolandgeider committed:
Properly handle the UUID for powersync body weight data

rolandgeider committed:
Simplify the powersync handlers by registering the individual ones

rolandgeider committed:
Simplify the powersync handlers by registering the individual ones

rolandgeider committed:
Make the license author non-nullable

This is more consistent with the rest of the license fields and makes handling
it a bit easier

rolandgeider committed:
Merge remote-tracking branch 'origin/master' into powersync

# Conflicts:
#	CHANGELOG.md
#	uv.lock
#	wger/manager/models/log.py
#	wger/manager/models/session.py

rolandgeider committed:
Fix test and linting

rolandgeider committed:
Read the PS_DATABASE_URI variable, if set

rolandgeider committed:
Merge remote-tracking branch 'origin/master' into powersync

rolandgeider committed:
Fix tests
